### DISCUSSION TITLE
不知道作者能不能出一套自定义表单控件？

### DISCUSSION THREAD

**Discussion by BoyFaceGirl** [iNKORE-NET/UI.WPF.Modern#97] [Ideas]
大部分wpf 的自定义控件库其实一言难尽，秀技的成分很多，都不是很实用，最常用的表单类控件封装反而比较少，之前找到一款 handorg 家的 https://github.com/HandyOrg/HandyControl 发现封装的表单控件还是比较使用，比如 textbox 加入了 title附加属性，这可太方便，尤其是制作erp类的商务软件，大量的各种表单元素，频繁的布局 标题和 各种表单控件，很麻烦，不知道作者有没有兴趣出一套类似的，仅仅包含基础表单的封装控件，textbox、number、combox、listview就增加 title标题 和 icon 以及 clear之类的，button 增加 loading icon 之类的

**Comment by NotYoojun** [comment]
emmm，我觉得这个可以有，不过的话我想说下列几个点：

1. UI.WPF.Modern 这个库是尽量挨着 WinUI 和 CommunityToolkit 来的，所有的组件和 API 也尽量朝它们对齐。自定义表单这个组件很明显并不是 WinUI 这一范畴里面的，所以我觉得要做的话还是单独开一个库比较好；

3. 本人最近把重心转移到 Web 和 Electron 开发上了，所以如果是我一个人来做的话还是有点不太现实，再加上因为私事这段时间都会忙家里的一些别的事情。所以更多的可能还是需要靠社区的力量。

**Reply by BoyFaceGirl** [reply]
> emmm，我觉得这个可以有，不过的话我想说下列几个点：
> 
> 1. UI.WPF.Modern 这个库是尽量挨着 WinUI 和 CommunityToolkit 来的，所有的组件和 API 也尽量朝它们对齐。自定义表单这个组件很明显并不是 WinUI 这一范畴里面的，所以我觉得要做的话还是单独开一个库比较好；
> 2. 本人最近把重心转移到 Web 和 Electron 开发上了，所以如果是我一个人来做的话还是有点不太现实，再加上因为私事这段时间都会忙家里的一些别的事情。所以更多的可能还是需要靠社区的力量。

哎，看来桌面开发真的是没落了，electron这个东西一言难尽

**Reply by NotYoojun** [reply]
@BoyFaceGirl 没办法，我们也是考虑到开发成本周期等等多重因素才艰难地做出这个决定。不过我们用 electron 会采用共享内核模式，多个产品共用同一个浏览器和 electron 内核。

不得不说，原生开发对于相当一部分中小企业来说成本还是太高了。😭😭
